src/components/MultiselectProblem.tsx

```diff
@@ -1,7 +1,118 @@
+  const [feedback, setFeedback] = useState('')
+  const [formDisabled, setFormDisabled] = useState(false)
+  const [retriesAllowed, setRetriesAllowed] = useState(0)
+  const solutionArray: string[] = JSON.parse(solution)
+  const parsedOptionValues: string[] = JSON.parse(solutionOptions)
+
+  const schema = Yup.object({
+    response: Yup.array().min(1, 'Please select an answer')
+  })
+
+  const contentRefCallback = useCallback((node: HTMLDivElement | null): void => {
+    if (node != null) {
+      mathifyElement(node)
+    }
+  }, [])
+
+  const clearFeedback = (): void => {
+    setFeedback('')
+  }
+
+  const modifyModel = (values: MultiselectFormValues, e: React.ChangeEvent<HTMLInputElement>): string[] => {
+    const newSet = new Set(values.response)
+    if (e.target.checked) {
+      newSet.add(e.target.value)
+    } else {
+      newSet.delete(e.target.value)
+    }
+    return Array.from(newSet)
+  }
+
+  const generateOptions = (values: MultiselectFormValues, isSubmitting: boolean, setFieldValue: (field: string, value: any, shouldValidate?: boolean) => void): JSX.Element[] => {
+    const options: JSX.Element[] = []
+
+    parsedOptionValues.forEach(val => options.push(
+    <label key={val}>
+      <Field
+      key={val}
+      type="checkbox"
+      name="response"
+      onChange={(e: React.ChangeEvent<HTMLInputElement>) => { clearFeedback(); setFieldValue('response', modifyModel(values, e)) }}
+      disabled={isSubmitting || formDisabled}
+      value={val}>
+      </Field>
+      {val}
+    </label>
+    ))
+
+    return options
+  }
+
+  const compareForm = (form: string[], solution: string[]): boolean => {
+    if (form.length !== solution.length) {
+      return false
+    }
+    for (let i = 0; i < solution.length; i++) {
+      if (!solution.includes(form[i])) {
+        return false
+      }
+    }
+    return true
+  }
+
+  const handleSubmit = async (values: MultiselectFormValues): Promise<void> => {
+    if (compareForm(values.response, solutionArray)) {
+      setFeedback(correctResponse)
+      solvedCallback()
+      setFormDisabled(true)
+    } else if (retryLimit === 0 || retriesAllowed !== retryLimit) {
+      setRetriesAllowed(currRetries => currRetries + 1)
+      setFeedback(encourageResponse)
+      allowedRetryCallback()
+    } else {
+      exhaustedCallback()
+      setFeedback(NO_MORE_ATTEMPTS_MESSAGE)
+      setFormDisabled(true)
+    }
+  }
 
-export const MultiselectProblem = ({ solvedCallback, exhaustedCallback, allowedRetryCallback }: MultiSelectProps): JSX.Element => {
-  return <></>
+  return (
+    <div className="os-raise-bootstrap" ref={contentRefCallback}>
+      <div className="my-3" dangerouslySetInnerHTML={{ __html: content }} />
+      <Formik
+        initialValues={{ response: [] }}
+        onSubmit={handleSubmit}
+        validationSchema={schema}
+      >
+        {({ isSubmitting, setFieldValue, values }) => (
+          <Form>
+            <div role="group">
```

Comment:
I took a brief look at the docs [here](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/group_role), but it's not immediately clear to me what this `role="group"` attribute does in practice. Here's what I see in the accessibility with and without it:

![image](https://user-images.githubusercontent.com/839025/168649519-042373a0-3d9f-4fc7-8427-93db6d5d1ee2.png)

![image](https://user-images.githubusercontent.com/839025/168649597-8244981d-2829-4f5a-9e97-d59260a7cfe8.png)

Do we know what the difference is from an accessibility / screen reader standpoint?

(cc: @christopher-mei in case you might have insights from your work)